test/WebJobs.Script.Tests.Integration/Host/ScriptHostManagerTests.cs

```diff
@@ -37,88 +37,119 @@ public ScriptHostManagerTests()
                         var blob2 = UpdateOutputName("first", "testblob", fixture).Result;
 
                         // wait for newly executed
-                        TestHelpers.Await(async () =>
-                        {
-                            bool exists = await blob2.ExistsAsync();
-                            return exists;
-                        }, timeout: 30 * 1000).Wait();
+                        TestHelpers.Await(() =>
+                         {
+                             bool exists = blob2.ExistsAsync().GetAwaiter().GetResult();
+                             return exists;
+                         }, timeout: 30 * 1000, userMessageCallback: GetErrorTraces).GetAwaiter().GetResult();
+
+                        // The TimerTrigger can fire before the host is fully started. To be more
+                        // reliably clean up the test, wait until it is running before calling Stop.
+                        TestHelpers.Await(() => manager.State == ScriptHostState.Running,
+                            userMessageCallback: GetErrorTraces).GetAwaiter().GetResult();
                     }
                     catch (Exception ex)
                     {
-                        exception = ExceptionDispatchInfo.Capture(ex);
+                        exceptions.Add(ex);
                     }
                     finally
                     {
                         try
                         {
-                            UpdateOutputName("first", "testblob", fixture).Wait();
+                            // Calling Stop (rather than using a token) lets us wait until all listeners have stopped.
+                            manager.Stop();
                         }
-                        catch
+                        catch (Exception ex)
                         {
+                            exceptions.Add(ex);
                         }
                     }
-
-                    cts.Cancel();
                 });
-                t.Start();
-
-                manager.RunAndBlock(cts.Token);
 
-                t.Join();
+                try
+                {
+                    backgroundThread.Start();
+                    manager.RunAndBlock();
+                    Assert.True(backgroundThread.Join(60000), "The background task did not complete in 60 seconds.");
 
-                Assert.True(exception == null, exception?.SourceException?.ToString());
+                    string exceptionString = string.Join(Environment.NewLine, exceptions.Select(p => p.ToString()));
+                    Assert.True(exceptions.Count() == 0, exceptionString);
+                }
+                finally
+                {
+                    // make sure to put the original names back
+                    await UpdateOutputName("first", "testblob", fixture);
+                }
             }
         }
 
         [Fact]
-        public async Task RenameFunctionAndRestart()
+        public void RenameFunctionAndRestart()
```

Comment:
Is there a reason for this change?

Reply:
I mangled the merge a bit. I've updated it.